src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -242,6 +242,107 @@
     [n.strip().upper() for n in snowflake_reserved_keywords.split("\n")]
 )
 
+# Add datetime units and their aliases from
+# https://docs.snowflake.com/en/sql-reference/functions-date-time.html#label-supported-date-time-parts
+snowflake_dialect.sets("datetime_units").clear()
```

Comment:
In #2025 for postgres I just added the datetime units that weren't in ansi, however, for snowflake there were so many that it made more sense to just include a complete list here. Makes sense that we match the documentation for the dialect if we have it 👍 

Do wonder if we should just do this for postgres too for completeness?